code/datums/weather/weather_types.dm

```diff
@@ -0,0 +1,115 @@
+//Different types of weather.
+
+/datum/weather/floor_is_lava //The Floor is Lava: Makes all turfs damage anyone on them unless they're standing on a solid object.
+	name = "the floor is lava"
+	desc = "The ground turns into surprisingly cool lava, lightly damaging anything on the floor."
+
+	telegraph_message = "<span class='warning'>Waves of heat emanate from the ground...</span>"
+	telegraph_duration = 150
+
+	weather_message = "<span class='userdanger'>The floor is lava! Get on top of something!</span>"
+	weather_duration_lower = 300
+	weather_duration_upper = 600
+	weather_overlay = "lava"
+
+	end_message = "<span class='danger'>The ground cools and returns to its usual form.</span>"
+	end_duration = 0
+
+	area_type = /area
+	target_z = ZLEVEL_STATION
+
+	overlay_layer = 2 //Covers floors only
+	immunity_type = "lava"
+
+/datum/weather/floor_is_lava/impact(mob/living/L)
+	for(var/obj/structure/O in L.loc)
+		if(O.density)
+			return
+	if(L.loc.density)
+		return
+	if(!L.client) //Only sentient people are going along with it!
+		return
+	L.adjustFireLoss(3)
+
+
+/datum/weather/advanced_darkness //Advanced Darkness: Restricts the vision of all affected mobs to a single tile in the cardinal directions.
+	name = "advanced darkness"
+	desc = "Everything in the area is effectively blinded, unable to see more than a foot or so around itself."
+
+	telegraph_message = "<span class='warning'>The lights begin to dim... is the power going out?</span>"
+	telegraph_duration = 150
+
+	weather_message = "<span class='userdanger'>This isn't your everday darkness... this is <i>advanced</i> darkness!</span>"
```

Comment:
LMFAO, Spongebob memes pls :star::fish:

Also a typo @Fox-McCloud: '**everday**' to '**everyday**'